library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/mapper/StringObfuscatorTest.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.recorder.mapper
+
+import com.datadog.android.sessionreplay.utils.ForgeConfigurator
+import com.datadog.tools.unit.extensions.ApiLevelExtension
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(ApiLevelExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(ForgeConfigurator::class)
+internal class StringObfuscatorTest {
+
+    lateinit var testedObfuscator: StringObfuscator
+
+    @BeforeEach
+    fun `set up`() {
+        testedObfuscator = StringObfuscator()
+    }
+
+    @Test
+    fun `M mask String W maskString(){string with newline}`(
+        forge: Forge
+    ) {
+        // Given
+        val fakeExpectedChunk1 = forge.aString(size = forge.anInt(1, max = 10)) { '\n' }
+        val fakeExpectedChunk2 = forge.aString { 'x' }
+        val fakeExpectedChunk3 = forge.aString(size = forge.anInt(1, max = 10)) { '\n' }
+        val fakeExpectedChunk4 = forge.aString { 'x' }
+        val fakeExpectedChunk5 = forge.aString(size = forge.anInt(1, max = 10)) { '\n' }
+        val n = forge.anInt(min = 1, max = 10)
+        val fakeExpectedText = (
+            fakeExpectedChunk1 +
+                fakeExpectedChunk2 +
+                fakeExpectedChunk3 +
+                fakeExpectedChunk4 +
+                fakeExpectedChunk5
+            ).times(n)
```

Comment:
why do we need `times` call if it just copies the same?

Reply:
to have a bigger text. I was following @xgouchet suggestion, not sure that this is what he wanted but I think is a bit overkill also. @xgouchet do you think this is still needed ?

Reply:
the `times` was more to generate a string with just 'x' of a given size, here it doesn't make sense